src/openapi/streaming/streaming.ts

```diff
@@ -923,7 +939,11 @@ class Streaming extends MicroEmitter<EmittedEvents> {
         this.subscriptions.push(subscription);
 
         // set the subscription to connection unavailable, the subscription will then subscribe when the connection becomes available.
-        if (this.connectionState !== this.CONNECTION_STATE_CONNECTED) {
+        // if shouldSubscribeBeforeStreamingSetup is set it will subscribe immediately
+        if (
+            !this.shouldSubscribeBeforeStreamingSetup &&

```

Comment:
By default subscription is created with connection availability flag set to true so now every logic inside a subscription object will think the connection is available if `shouldSubscribeBeforeStreamingSetup` is set to true even if we haven't connected yet. Have you checked if it's valid to make a subscription subscribe before we connect just by telling the whole machinery inside a subscription we are connected even if we're not?

Reply:
yes this is what this feature is about,

- so we will initially have this flag set to true so all subscription can subscribe immediately i.e. they can send the POST request to publisher and can get the initial snapshot response ,so by this we can render our UI early even though streaming connection hasn't setup yet

- later on when streaming connection establish we will receive the deltas through the streaming connection.

- regarding the flag _shouldSubscribeBeforeStreamingSetup_ we initally set it to true so all subscriptions can immediately subscribe and later on if any disconnect/fail/reconnect happens we set it to false, i.e. go back to the normal flow